sqlx-cli/src/database.rs

```diff
@@ -58,26 +60,34 @@ pub async fn setup(migration_source: &str, connect_opts: &ConnectOpts) -> anyhow
     migrate::run(migration_source, connect_opts, false, false, None).await
 }
 
-fn ask_to_continue_drop(db_url: &str) -> bool {
-    loop {
-        let r: Result<String, ReadlineError> =
-            prompt(format!("Drop database at {}? (y/n)", style(db_url).cyan()));
-        match r {
-            Ok(response) => {
-                if response == "n" || response == "N" {
-                    return false;
-                } else if response == "y" || response == "Y" {
-                    return true;
-                } else {
-                    println!(
-                        "Response not recognized: {}\nPlease type 'y' or 'n' and press enter.",
-                        response
-                    );
-                }
-            }
-            Err(e) => {
-                println!("{e}");
-                return false;
+async fn ask_to_continue_drop(db_url: String) -> bool {
+    let ctrlc_fut = signal::ctrl_c();
+    let decision_fut = task::spawn_blocking(move || {
+        Confirm::new()
+            .with_prompt(format!("Drop database at {}?", style(&db_url).cyan()))
+            .wait_for_newline(true)
+            .default(false)
+            .show_default(true)
+            .interact()
+    });
+
+    let restore_cursor = || Term::stderr().show_cursor().unwrap();
+
+    select! {
+        biased;
```

Comment:
Context: despite `biased` sometimes the CTRL + C future returns first, sometimes the decision futures is ready first.

Reply:
Thinking more about it maybe it's simply the outer CTRL + C catcher that is causing the problem. I'll try later.

Reply:
Fixed it. Now it works reliably for me